fava/static/javascript/editor.js

```diff
@@ -254,6 +255,26 @@ export default function initSourceEditor(name) {
       });
     }
   });
+
+  const leaveMessage = 'There are unsaved changes. Are you sure you want to leave?';
+
+  router.on('navigate', () => {
+    if (!editor.getDoc().isClean()) {
+      const leave = confirm(leaveMessage); // eslint-disable-line no-restricted-globals,no-alert
+      if (leave) {
+        editor.getDoc().markClean();
+      }
+      return !leave;
+    }
+    return false;
+  });
+
+  window.onbeforeunload = () => {
```

Comment:
This and the other event handler persist when one navigates away from the editor. Better use window.addEventListener and add the event handlers once globally (on page-init)

Reply:
The check if the editor is dirty needs a reference to the `editor`-variable, pointing to the CodeMirror instance running the editor. I don't know how to get this reference when putting this code in a global scope without using a global variable (or is that what you suggest?).

Reply:
This doesn't need a global variable, one within the scope of this file is enough. Right now the above attaches tons of event handlers even though at most one is needed